### PR TITLE
[PAY-406] Web: Add Coinbase Pay Button

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds and integrates the Coinbase Pay button. See https://github.com/coinbase/cbpay-js

Not wired up to anything, just creates the component.

Usage:
```ts

const SomeComponent = () => {
  const coinbasePay = useContext(CoinbasePayContext)
  return (
    <CoinbasePayButton
        size={CoinbasePayButtonSize.COMPACT}
        variant={CoinbasePayButtonVariant.BUY}
        resolution={CoinbasePayButtonImageResolution.X2}
        onClick={() => {
          coinbasePay.resetParams({
            presetFiatAmount: 3.50, // set to some amount
            onSuccess: () => {
              // do something
          })
          coinbasePay.open()
        }}
        isDisabled={!estimatedUSDC || !coinbasePay.isReady}
      />
    )
}


<CoinbasePayButtonProvider>
  //...
  <SomeComponent />
</CoinbasePayButtonProvider>
```

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

- ~~Uses `require.context` to lazily load the branded Coinbase images, since we won't need most (or nearly all) of them until absolutely necessary. This should split the bundle and let bundle size remain small despite leaving the option to specify different image variants and staying within the bounds of the mandated images.~~
- ~~Doesn't work on every domain - needs specific access. If it fails to initialize, the button stays disabled.~~
- ~~Defaults to "popup" experience. Might have to test further to ensure popup blockers don't block it. (Note: this is another good reason to stay away from Sagas - popup blockers tend to not block popups triggered within an event handler for user interaction)~~

- Coinbase Pay instance might not be ready when open is called? In testing this didn't seem to have any impact.
- Should probably disable other usage of the button when the experience is open?

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

~~Tested using my Phantom wallet on https://marcus.audius.co~~
Tested on marcus.audio.co E2E with a user bank wallet

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

No monitoring. Probably need to think more here about how to know what's happening when things fail.

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

No feature flags necessary (yet) since the button isn't wired to anything

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cb-pay-button

File: packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        closeOnSuccess: true
+      })
+    } else {
+      setIsReady(false)
+    }
+    return () => cbInstance.current?.destroy()
+  }, [destinationWalletAddress, amount, cbInstance, onExit, onSuccess])
+
+  // Lazy load the image
+  useEffect(() => {
```
Comment (dylanjeffers) [inline packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx:139]: maybe useAsync here?

File: packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  destinationWalletAddress?: string
+  amount?: number
+  onSuccess?: () => void
+  onExit?: () => void
+}) => {
+  const [isReady, setIsReady] = useState(false)
+  const [imageSrc, setImageSrc] = useState<string>()
+  const cbInstance = useRef<ReturnType<typeof initOnRamp>>()
+
+  useEffect(() => {
```
Comment (dylanjeffers) [inline packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx:104]: this okay potentially getting called multiple times?

**Review by dylanjeffers — APPROVED** [review]
awesome finds and making this work

File: packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx:104]: It should get cleaned up (see returned function) so I think so

File: packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx (hunk 1 above)
Comment (rickyrombo) [reply packages/web/src/components/coinbase-pay-button/CoinbasePayButton.tsx:139]: 🤯

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cb-pay-button

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cb-pay-button

**Review by piazzatron — APPROVED** [review]
Sweet, much prefer the CB provider pattern! Looks great

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cb-pay-button

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cb-pay-button

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cb-pay-button

**Pull request merged**